java-vtl-script/src/main/java/no/ssb/vtl/script/functions/VTLSubstr.java

```diff
@@ -65,26 +65,22 @@ protected VTLString safeInvoke(TypeSafeArguments arguments) {
             );
         }
 
-        int substringLength = value.get().length() - 1 - startPosition.get().intValue();
-        if (substringLength < 0) {
+        if (startPosition.get().intValue() >= value.get().length()) {
             //deviation from the VTL specification 1.1
             return VTLString.of("");
         }
 
-        if (ignoreLengthArgument(startPosition.get(), length.get(), value.get())) {
-            return VTLString.of(value.get().substring(startPosition.get().intValue()));
-        } else {
-            Long endIndex = startPosition.get() + length.get();
-            return VTLString.of(value.get().substring(startPosition.get().intValue(), endIndex.intValue()));
-        }
+        return VTLString.of(value.get().substring(
+                startPosition.get().intValue(),
+                calculateEndPosition(startPosition.get().intValue(), length.get(), value.get().length())));
```

Comment:
Awesome!
Maybe just extract `endPosition`  as a variable to make it a little easier to read...?